2-structured-data/src/app.php

```diff
@@ -40,6 +43,35 @@
 
 $app['config'] = Yaml::parse(file_get_contents($config));
 
+// determine the datamodel backend using the app configuration
+$app['bookshelf.model'] = function ($app) {
+    /** @var array $config */
+    $config = $app['config'];
+
+    // Data Model
+    switch ($config['bookshelf_backend']) {
```

Comment:
Nit: it will throw Undefined Index when 'bookshelf_backend' is missing. It might not be a big deal since we have config template, but it is still possible you accidentally change that key (I have done such things many times).